pkg/console/validator.go

```diff
@@ -0,0 +1,106 @@
+package console
+
+import (
+	"fmt"
+	"net"
+
+	"github.com/pkg/errors"
+	"github.com/rancher/harvester-installer/pkg/config"
+	"github.com/sirupsen/logrus"
+)
+
+var (
+	ErrMsgModeCreateContainsServerURL   = fmt.Sprintf("ServerURL need to be empty in %s mode", modeCreate)
+	ErrMsgModeJoinServerURLNotSpecified = fmt.Sprintf("ServerURL can't empty in %s mode", modeJoin)
+	ErrMsgModeUnknown                   = "Unknown mode"
+	ErrMsgTokenNotSpecified             = "token not specified"
+
+	ErrMsgMgmtInterfaceNotSpecified = "no management interface specified"
+	ErrMsgMgmtInterfaceNotFoud      = "interface not found"
+	ErrMsgMgmtInterfaceIsLoop       = "interface is a loopbck interface"
+	ErrMsgDeviceNotSpecified        = "no device specified"
+	ErrMsgDeviceNotFound            = "device not found"
+	ErrMsgNoCredentials             = "no SSH authorized keys or passwords are set"
+)
+
+type ValidatorInterface interface {
```

Comment:
These two methods not really like exported general functions for interface. I would suggest to do the below way, then it should be better for adding other validators in the future. WDYT?

type Validator interface {
  Validate() error
}

type HostValidator struct {} which has internal functions like checkMgmtInterface and checkDevice being called in Valdiate.

Reply:
Fixed!